### DISCUSSION TITLE
Keeping plugins up to date for future releases of Grails (6 and 7)

### DISCUSSION THREAD

**Discussion by ppazos** [apache/grails-core#13526] [Ideas]
Thinking of Grails 7 but in general too, is that when a major version of Grails is released, many plugins become unusable.

I know some core contributors took some popular plugins and keep maintain them updated, but most are one-time developers that are not maintaining them. The plugin portal is also outdated, and it's difficult to find plugins compatible with a specific version of Grails (there is no filter by version https://grails.org/plugins.html).

I don't know if there is a migration guide that allow to update plugins to new APIs, or a way to automatically notify plugin maintainers about a new major version release to keep their plugins up to date. It could even be a blocker for upgrading Grails because a plugin is crucial for the business (for instance I can't live without the mail plugin).

Just wanted to mention that the process behind plugins is also part of keeping the framework relevant and the community happy.

**Comment by gsartori** [comment]
We have the same issue with the Websocket plugin we rely upon. We may be able to do what we are doing just using the Spring API avoiding the dependency to that plugin. I will personally try not to depend on plugins for core parts of our software (but that means we are going to create our own plugin so maybe it’s better to spend time upgrading the plugin itself).

Anyway, I would try to solve the “unmaintained plugins” problem with marketing. We could try to raise the Grails hype again so to inform the maintainers that the project is worth investing your time on.

Another (maybe better) option could be having a list of “Official plugins” we need to maintain. A marketing action like “Adopt a plugin”.

A smaller piece of software may be something each one of us working with Grails could take responsibility for. It’s a way of contributing to the project.

**Comment by sbglasius** [comment]
We have the Grails Plugin Collective https://github.com/gpc, which is mainly maintained by yours truly, but as I have limited bandwidth, I can not do it on my own. 
The steering group is talking about moving more plugins to the GPC, and I'm fine with that, as long as others help maintain them.

**Reply by ppazos** [reply]
That's great. Is GPC under the Grails Foundation? If not, it would be strategic to formalize some kind of formal relationship.

**Comment by jdaugherty** [comment]
FYI: as we've upgraded plugins for grails-core, we have realized there are several "best practices" to make plugins future compatible.  I've created https://github.com/apache/grails-core/issues/14129 to start to capture some of this information and eventually we'll update the docs from this ticket.
